src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -325,8 +325,7 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
             // Try graceful shutdown with q() first.
             if (host != null) {
                 try {
-                    host.QuitAsync().SilenceException<Exception>().DoNotWait();
-                    await Task.WhenAny(hostRunTask, Task.Delay(2000)).Unwrap();
+                    host.QuitAsync().SilenceException<Exception>().Wait(10000);
```

Comment:
There seem to be more than one caller of this method. Do we want to block in one case and not the other?